### PR TITLE
Update Canal and Calico VXLAN to v3.23.3

### LINKED ISSUE #2118: Update Calico/Canal to v3.23
### Description of the feature you would like to add / User story

We should update Calico/Canal to the latest minor release (v3.22) in order to get the latest features and bug fixes.

### Solution details

- Update Calico in the Canal addon to v3.23
- Update Calico VXLAN addon to v3.23

### PR DESCRIPTION
**Pull request by xmudrii**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

This PR updates Canal and Calico VXLAN to the latest available version (v3.23.3).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2118 

**Does this PR introduce a user-facing change?**:
```release-note
Update Canal and Calico VXLAN to v3.23.3
```

/assign @ahmedwaleedmalik

### CODE REVIEW

File: addons/cni-canal/canal.yaml [hunk 1]
```diff
@@ -4341,7 +4483,7 @@ spec:
             privileged: true
           resources:
             requests:
-              cpu: {{ default .Params.RequestsCPU "250m" }}
+              cpu: 250m
```
Comment (ahmedwaleedmalik) [inline addons/cni-canal/canal.yaml:4486]: Please revert this change. 
```suggestion
             cpu: {{ default .Params.RequestsCPU "250m" }}
```

File: addons/calico-vxlan/calico-vxlan.yaml [hunk 2]
```diff
@@ -1263,12 +1364,22 @@ spec:
...
               vxlanEnabled:
+                description: 'VXLANEnabled overrides whether Felix should create the
+                  VXLAN tunnel device for VXLAN networking. Optional as Felix determines
+                  this based on the existing IP pools. [Default: nil (unset)]'
                 type: boolean
               vxlanMTU:
-                description: 'VXLANMTU is the MTU to set on the tunnel device. See
-                  Configuring MTU [Default: 1440]'
+                description: 'VXLANMTU is the MTU to set on the IPv4 VXLAN tunnel
+                  device. See Configuring MTU [Default: 1410]'
```
Comment (xmudrii) [inline addons/calico-vxlan/calico-vxlan.yaml:1378]: It appears that the default MTU has been changed in Calico 3.23. Do we need to take any action on this?

File: addons/calico-vxlan/calico-vxlan.yaml [hunk 3]
```diff
@@ -4316,6 +4446,9 @@ spec:
             # Enable or Disable VXLAN on the default IP pool.
             - name: CALICO_IPV4POOL_VXLAN
               value: "CrossSubnet"
+            # Enable or Disable VXLAN on the default IPv6 IP pool.
+            - name: CALICO_IPV6POOL_VXLAN
+              value: "CrossSubnet"
```
Comment (xmudrii) [inline addons/calico-vxlan/calico-vxlan.yaml:4451]: Since we don't support dual-stack or IPv6, should we disable this option?

File: addons/cni-canal/canal.yaml [hunk 4]
```diff
@@ -1280,12 +1381,22 @@ spec:
...
               vxlanEnabled:
+                description: 'VXLANEnabled overrides whether Felix should create the
+                  VXLAN tunnel device for VXLAN networking. Optional as Felix determines
+                  this based on the existing IP pools. [Default: nil (unset)]'
                 type: boolean
               vxlanMTU:
-                description: 'VXLANMTU is the MTU to set on the tunnel device. See
-                  Configuring MTU [Default: 1440]'
+                description: 'VXLANMTU is the MTU to set on the IPv4 VXLAN tunnel
+                  device. See Configuring MTU [Default: 1410]'
```
Comment (xmudrii) [inline addons/cni-canal/canal.yaml:1395]: Same here -- do we need to take any action on this MTU change?

File: addons/cni-canal/canal.yaml (hunk 1 above)
Comment (xmudrii) [reply addons/cni-canal/canal.yaml]: Fixed.

File: addons/calico-vxlan/calico-vxlan.yaml (hunk 2 above)
Comment (ahmedwaleedmalik) [reply addons/calico-vxlan/calico-vxlan.yaml:1378]: I don't think so.

Comment (kron4eg) [reply addons/calico-vxlan/calico-vxlan.yaml:1378]: I wonder why they did it

Comment (kron4eg) [reply addons/calico-vxlan/calico-vxlan.yaml:1378]: It's about ipv6 support: https://github.com/projectcalico/calico/commit/009498285c7f6634ab0fb3db7b38c4567f4cbe34

So probably we can do the same

File: addons/cni-canal/canal.yaml (hunk 4 above)
Comment (rastislavs) [reply addons/cni-canal/canal.yaml:1395]: I think for Canal this effectively isn't important at all - not used, as Calico is used only to provide network policies support, VXLANs are done by Flannel

File: addons/calico-vxlan/calico-vxlan.yaml (hunk 3 above)
Comment (rastislavs) [reply addons/calico-vxlan/calico-vxlan.yaml:4451]: I think that in case that IPv6 is not enabled (there are no IPv6 pools) this should not be important (should have no effect). Since `CALICO_IPV4POOL_VXLAN` is set to `CrossSubnet` as well, I think it is dafe to just keep it like it is.

File: addons/calico-vxlan/calico-vxlan.yaml (hunk 2 above)
Comment (rastislavs) [reply addons/calico-vxlan/calico-vxlan.yaml:1378]: To me this actually isn't a change for us - it is a change in the `FelixConfiguration` CRD, that we are not using, as we do not deploy Calico via operator. In this manifest-based deployment of Calico, there is no use of `FelixConfiguration` at all, and the MTU is passed from the `veth_mtu` config in `calico-config` configmap. By default is set to `0` - which means auto-configuration - and that should work for most of the deployments.

**Comment by xmudrii** [comment]
/override pull-kubeone-e2e-aws-defaults-calico-containerd-v1.22.11
It seems like this is not caused by this update, so I created #2192 to track and fix this issue.
